src/Language/Haskell/Ghcid.hs

```diff
@@ -118,11 +119,14 @@ startGhciProcess process echo0 = do
                 res2 <- onceFork $ consume Stderr (finish Stderr)
                 res1 <- res1
                 res2 <- res2
-                case liftM2 (,) res1 res2 of
-                    Nothing -> case cmdspec process of
-                        ShellCommand cmd -> throwIO $ UnexpectedExit cmd msg
-                        RawCommand exe args -> throwIO $ UnexpectedExit (unwords (exe:args)) msg
-                    Just v -> return v
+                case (res1, res2) of
+                    (Right v1, Right v2) -> return (v1, v2)
+                    (_, Left err) -> case cmdspec process of
```

Comment:
This bit is a tag duplicative. Can we `let raise msg err = throwIO $ case cmdspec process of ShellCommand cmd -> UnexpectedExit cmd msg err; ...` or similar? Anything that stops the logic being repeated twice.